core/src/main/java/hudson/model/Run.java

```diff
@@ -659,20 +679,42 @@ public final long getStartTimeInMillis() {
         return startTime;
     }
 
+    //TODO(oleg_nenashev): Do we want to export full description by default taking the size considerations? Looks like it should be another REST API or a flag
```

Comment:
I wonder whether @harry-g and @jsoref use REST APIs for builds. The current state is a potentially big problem for long descriptions

Reply:
Why would anything change for REST API? `getDescription` always returned full text.

Reply:
That's why I do not touch it. But I believe that it is not ideal for users who are concerned about huge REST APIs

Reply:
No, I think we don't use it afaik. 
But I would also discourage delivering the full description with the build object.
(Where it is only one build run, so less severe than the current job/build with/config page which load description from tons of builds.)

Reply:
Just decrease visibility if you don't like it here. People should be requesting it via `?tree` anyway.

More of a separate change though.